src/field_theory/splitting_field.lean

```diff
@@ -739,3 +739,89 @@ end is_splitting_field
 end splitting_field
 
 end polynomial
+
+section alg_equiv_restrict
+
+open polynomial
+
+variables {F K : Type*} [field F] [field K] [algebra F K] (ϕ ψ : K →ₐ[F] K) (χ ω : K ≃ₐ[F] K)
+  (p : polynomial F) (E : Type*) [field E] [algebra F E] [algebra E K]  [is_scalar_tower F E K]
+
+lemma is_scalar_tower_to_alg_hom_range_eq_adjoin [hp : is_splitting_field F E p] :
+  (is_scalar_tower.to_alg_hom F E K).range =
+    algebra.adjoin F (↑(p.map (algebra_map F K)).roots.to_finset : set K) :=
+begin
+  rw [is_scalar_tower.algebra_map_eq F E K, ←map_map, roots_map, ←finset.image_to_finset,
+    finset.coe_image, algebra.adjoin_algebra_map, hp.adjoin_roots, algebra.map_top],
+  exact ((splits_id_iff_splits (algebra_map F E)).mpr hp.splits),
+end
```

Comment:
This is an interesting lemma because there are a few choices that (if mathlib were much more complete?) might be made differently:
 * the `subalgebra`-valued range of `is_scalar_tower.to_alg_hom` is in principle the same set as the range of `algebra_map`, or the `field_range` of `algebra_map`, or a hypothetical `intermediate_field` range of `to_alg_hom`. I'm assuming you went for the `subalgebra` because you need an `algebra` instance below and there is no easy way to turn it into an `intermediate_field` yet, correct?
 * similarly, `algebra.adjoin` is the same set as `intermediate_field.adjoin` since each root is algebraic. Was the choice for `subalgebra` inspired by the above decision?

Reply:
Yes, I think that at some point the `splitting_field` file should be refactored to use `intermediate_field.adjoin`. But for now, talking about an equality of subalgebras seems to be the cleanest way to state things (especially when considering the proof in the following definition).